localstack/services/sqs/sqs_listener.py

```diff
@@ -58,6 +59,25 @@ def forward_request(self, method, path, data, headers):
 
         return True
 
+    # Format of the message attribute is MessageAttribute.<int id>.<field>
+    def format_message_attributes(self, data):
+        names = []
+        for (k, name) in [(k, data[k]) for k in data if k.startswith('MessageAttribute') and k.endswith('.Name')]:
+            attr_name = name[0]
+            k_id = k.split('.')[1]
+            names.append((attr_name, k_id))
+
+        msg_attrs = {}
+        for (key_name, key_id) in names:
+            msg_attrs[key_name] = {}
+            # Find vals for each key_id
+            attrs = [(k, data[k]) for k in data
+                if k.startswith('MessageAttribute.{}.'.format(key_id)) and not k.endswith('.Name')]
+            for (attr_k, attr_v) in attrs:
+                msg_attrs[key_name][attr_k.split('.')[3]] = attr_v[0]
```

Comment:
Should this be `attr_k.split('.')[2]` instead of `attr_k.split('.')[3]`?

Reply:
@whummer - The format for the Name field is `MessageAttribute.<id>.Name` while the field for the values is `MessageAttribute.<id>.Value.<Type>Value` and `MessageAttribute.<id>.Value.DataType`, so we really just need to the last field from that string.  The attributes are formatted correctly in what's passed to Lambda.  :)

Please let me know if you have more questions!

Reply:
Ah, ok makes sense - thanks for clarifying!